bridges/darwinia-ethereum/bridge/src/service/extrinsics/handler.rs

```diff
@@ -238,3 +244,48 @@ impl ExtrinsicsHandler {
         Ok(())
     }
 }
+
+impl ExtrinsicsHandler {
+    pub fn collect_message(&self, message: &Extrinsic) -> color_eyre::Result<()> {
+        let mut key: String = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_nanos()
+            .to_string();
+
+        if let Ok(true) = self.message_kv.exists(&key) {
+            let random: String = thread_rng()
+                .sample_iter(&Alphanumeric)
+                .take(1)
+                .map(char::from)
+                .collect();
+            key += &random;
+        }
+        self.message_kv.put(key, message)?;
+        Ok(())
+    }
+
+    pub async fn consume_message(&self) -> color_eyre::Result<()> {
+        loop {
+            let mut extrinsics = self.message_kv.sorted_keys()?;
+            if extrinsics.is_empty() {
+                tokio::time::sleep(std::time::Duration::from_secs(3)).await;
+                continue
+            }
+            while let Some(key) = extrinsics.pop() {
+                let ex: Extrinsic = self.message_kv.get_as_unwrap(&key)?;
+                match self.send_extrinsic(ex.clone()).await {
+                    Ok(_) => self.message_kv.delete(&key)?,
+                    Err(err) => {
```

Comment:
Same issue here #404

Reply:
I think, just this is enought, @furoxr  please update this.

Reply:
Update finished